### ISSUE TITLE
Tests not running after upgrading to 4.0.2-snapshot..

### ISSUE THREAD

**Issue by neo81-83** [prashant-ramcharan/courgette-jvm#137]
Hello Prashant,

I had some time to spend over the weekend and thought of trying your latest version which has report portal integration. After making necessary changes as suggested in the documentation, I ran tests but to my surprise none of the tests got launched, however there were no errors thrown either.

`-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.xyz.cucumber.runner.TestScenarioRunner
Configuring TestNG with: org.apache.maven.surefire.testng.conf.TestNG652Configurator@51016012
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 7.842 sec

Results :

Tests run: 1, Failures: 0, Errors: 0, Skipped: 0`

I suspect it is the runner class which is causing problems, as in version 3.3.0 I was using `import cucumber.api.CucumberOptions;` and things were perfectly fine BUT in the new version 4.0.2-snapshot it seems that you have wrapped CucumberOptions in to a custom class  `import courgette.api.CucumberOptions;` and there's no way of bypassing this. Let me know if you have any workaround available.

PFA the runner classes of old and new versions. Hope to hear soon.

Cheers!

[CourgetteRunner.docx](https://github.com/prashant-ramcharan/courgette-jvm/files/3855266/CourgetteRunner.docx)

**Comment by prashant-ramcharan** [comment]
Hello,

Thanks for reporting. I will look into this.

**Comment by prashant-ramcharan** [comment]
I had a quick look and tests are running fine.

If you running tests in the IDE, make sure your run settings are correct.

If you using a TestNG Courgette runner then you should run the test as a TestNG test.

<img width="1310" alt="intellij_settings" src="https://user-images.githubusercontent.com/2563149/69008126-beaf4000-093e-11ea-9b2d-c22c6d9d53b5.png">

<img width="725" alt="runner_testng" src="https://user-images.githubusercontent.com/2563149/69008128-c373f400-093e-11ea-8373-4c9fd63de03c.png">

**Comment by prashant-ramcharan** [comment]
Please let me know if you experience the same with the Courgette JVM example project.

https://github.com/prashant-ramcharan/courgette-jvm-example

**Comment by neo81-83** [comment]
Appreciate your prompt response that too on a weekend, I am running tests from command line and  when I used `-Dcucumber.tags` instead of `-Dcucumber.options` for passing the cucumber tags some action happened, i.e. screen full of exceptions.
```
SEVERE: Exception while executing pickle
java.util.concurrent.ExecutionException: io.cucumber.java.InvalidMethodSignatureException: A method annotated with Before, After, BeforeStep or AfterStep must have one of these signatures:
 * public void before_or_after(io.cucumber.java.Scenario scenario)
 * public void before_or_after()
at com.xyz.cucumber.stepdefs.CommonStepDefs.initialize(Scenario) in file:/Users/xyz/workspace/auto-dev/automation/target/test-classes/
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at io.cucumber.core.runtime.Runtime.run(Runtime.java:110)
	at io.cucumber.core.cli.Main.run(Main.java:63)
	at io.cucumber.core.cli.Main.main(Main.java:27)
Caused by: io.cucumber.java.InvalidMethodSignatureException: A method annotated with Before, After, BeforeStep or AfterStep must have one of these signatures:
 * public void before_or_after(io.cucumber.java.Scenario scenario)
 * public void before_or_after()
```
I am using `io.cucumber.java8` as my stepdefs are lambda expressions, but as per the exception before and after hooks are only accepting `io.cucumber.java`. So I changed a couple of stepdefs file to use `io.cucumber.java` after that I got new exception:
```
io.cucumber.core.exception.CucumberException: Step [I login to the console as '(.*?)'] is defined with 1 parameters at 'CommonStepDefs.java:56'.
However, the gherkin step has 0 arguments.
Step text: I login to the console as 'siteAdmin'
```
Pretty much stuck, don't know what to do :(

**Comment by prashant-ramcharan** [comment]
Your step definition should be:

````java
When("I login to the console as {string}", (String something) -> {
    // Write your code here
});

````

**Comment by neo81-83** [comment]
I have been using stepdefs like the one mentioned below with 3.3.0 without any issues, something in 4.0.2 is causing problems.
```
Given("I login to the console as '(.*?)'", (String userType) -> {
            Login loginPg = PageFactory.initElements(driver, Login.class);
            loginPg.login(userType);
        });
```

**Comment by prashant-ramcharan** [comment]
Courgette does not  and will not change the way Cucumber works under the hood.

Courgette version 4.0.2 is using Cucumber version 5.0.0-RC. This version upgrade means you have to clearly define Regular expression or Cucumber Expressions.  This is not something that Courgette has defined or enforced in version 4.0.2.

You can still use your step definitions but you have to clearly define it like:

````java
// The step definition must start with ^ and end with $
Given("^I login to the console as (.*?)$", (String userType) -> { ... });
````

Some references here:
https://cucumber.io/docs/cucumber/step-definitions/#expressions
https://cucumber.io/docs/cucumber/cucumber-expressions/

**Comment by neo81-83** [comment]
Thanks Prashant, step definition was the problem here, at least tests are running now. But results are not getting posted to reportportal, see the exception below:
```
java.lang.NoClassDefFoundError: org/apache/http/conn/ssl/TrustAllStrategy
	at courgette.integration.reportportal.ReportPortalService.sendMultiPartPost(ReportPortalService.java:53)
	at courgette.integration.reportportal.ReportPortalService.publishReport(ReportPortalService.java:37)
	at courgette.runtime.CourgetteRunner.publishReportToReportPortal(CourgetteRunner.java:165)
	at courgette.api.testng.TestNGCourgette.parallelRun(TestNGCourgette.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:816)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1124)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:774)
	at org.testng.TestRunner.run(TestRunner.java:624)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:359)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:354)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:312)
	at org.testng.SuiteRunner.run(SuiteRunner.java:261)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1191)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1116)
	at org.testng.TestNG.run(TestNG.java:1024)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:77)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:110)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:106)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.lang.ClassNotFoundException: org.apache.http.conn.ssl.TrustAllStrategy
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 37 more
Caused by: java.lang.ClassNotFoundException: org.apache.http.conn.ssl.TrustAllStrategy
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 37 more
```

**Comment by prashant-ramcharan** [comment]
Can you confirm that you have refreshed your project dependancies?

**Comment by neo81-83** [comment]
Yes, it has been refreshed and I was able to run a test successfully BUT integration with report portal seems to fail:
![image](https://user-images.githubusercontent.com/56172687/69014604-dba53c80-0959-11ea-8bda-e13b97bad77b.png)

**Comment by prashant-ramcharan** [comment]
Can you try adding this dependancy to your Pom.xml and let me know if it resolves it?

````maven
<dependency>
 <groupId>org.apache.httpcomponents</groupId>
    <artifactId>httpclient</artifactId>
    <version>4.5.10</version>
</dependency>
````

**Comment by neo81-83** [comment]
Thanks that resolved the issue, now i am able to post results to report portal:
![image](https://user-images.githubusercontent.com/56172687/69015203-c8956b00-095f-11ea-9948-b293fd0540b2.png)

How can `courgette_test_execution.xml #1` and `Test Suite` names be customized as per our requirements?

**Comment by prashant-ramcharan** [comment]
These are the default values as defined in Courgette when posting to ReportPortal.

I will make this configurable in the next release.

**Comment by prashant-ramcharan** [comment]
@neo81-83 This is now released in version `4.0.3-snapshot`
